src/Features/Core/Portable/DocumentHighlighting/AbstractDocumentHighlightsService.cs

```diff
@@ -297,9 +296,13 @@ private static bool ShouldIncludeDefinition(ISymbol symbol)
                     return !((INamedTypeSymbol)symbol).IsScriptClass;
 
                 case SymbolKind.Parameter:
-                    // If it's an indexer parameter, we will have also cascaded to the accessor
-                    // one that actually receives the references
-                    if (symbol.ContainingSymbol is IPropertySymbol containingProperty && containingProperty.IsIndexer)
+
+                    // If it's an indexer parameter, we will determine by the property 
+                    // symbols instead of the accessor symbols because accessor 
+                    // symbols with no references are filtered out
+                    if (symbol.ContainingSymbol is IMethodSymbol methodSymbol
+                        && methodSymbol.AssociatedSymbol is IPropertySymbol containingProperty
+                        && containingProperty.IsIndexer)
                     {
                         return false;
                     }
```

Comment:
From my understanding, if we remove the block, in certain cases the definition item will be added to the list of highlighted spans twice. In the case where parameter symbol has both accessor methods and property methods, it would be added twice.

Reply:
Can you give an example of that, and why it would be undesirable?

Reply:
@CyrusNajmabadi The concern was if we tagged a span twice would it end up getting formatted funny, or would the "go to next highlighted reference" or "go to previous highlighted reference" somehow get "stuck" if there were duplicated elements. We weren't sure if something actually deduplicates.